### PR TITLE
chore: replace @storybook/semver with semver

### PR DESCRIPTION
**Pull request by danielamram**
Issue: https://github.com/storybookjs/storybook/issues/19197

## What I did
- remove @storybook/semver in every`package.json` in the repo & add `sermver` instead
- replace all `@storybook/semver` usage references with `semver`
- remove `@storybook/semver` declarations from `typings.d.ts` and `verdaccio.yaml`
- [fix boolean type with `!!` following the new package](https://github.com/storybookjs/storybook/compare/next...danielamram:storybook:replace-storybook-semver?expand=1#diff-098d673697506bc0b8365f20a57f4202acf64111b1615eca22d26249478cd051R60)

## How to test
Tested via yarn 1 & yarn 2 - all packages built & unit tests passed successfully

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/frameworks/angular/package.json [hunk 1]
```diff
@@ -45,7 +45,6 @@
     "@storybook/csf": "0.0.2--canary.0899bb7.0",
     "@storybook/docs-tools": "7.0.0-alpha.34",
     "@storybook/node-logger": "7.0.0-alpha.34",
-    "@storybook/semver": "^7.3.2",
```
Comment (danielamram) [inline code/frameworks/angular/package.json:48]: add missing semver dependency

Comment (JReinhold) [reply code/frameworks/angular/package.json:48]: you're right, I think it needs to be here.

File: code/lib/ui/package.json [hunk 2]
```diff
@@ -99,5 +98,9 @@
       "./src/runtime.ts"
     ]
   },
-  "gitHead": "fc90fc875462421c1faa35862ac4bc436de8e75f"
+  "gitHead": "fc90fc875462421c1faa35862ac4bc436de8e75f",
+  "dependencies": {
```
Comment (JReinhold) [inline code/lib/ui/package.json:102]: These should be `devDependencies`

**Review by JReinhold — CHANGES_REQUESTED** [review]
Looks really good, only a few minor tweaks needed as you've also pointed out. 💪 

`@storybook/semver` is still listed a few places in `yarn.lock`s, but it looks like that problem will fix itself once `@storybook/api` gets released again with these changes, so it's fine. It doesn't get installed anyway.

File: code/lib/ui/package.json (hunk 2 above)
Comment (danielamram) [reply code/lib/ui/package.json]: Fixed ✅ 
Just wondering, why aren't u using `devDependencies` convention on other packages in the repo as well (it used only here)?

**Comment by danielamram** [comment]
Yo @JReinhold, [CircleCI pipelines are failing](https://app.circleci.com/pipelines/github/storybookjs/storybook/29530/workflows/3b8ede17-fb29-431b-a661-49101e396198/jobs/417240?invite=true#step-103-13), seems like it requiring for `@storybook/semver` package for some reason (cache?)
I'v been running `yarn task --task build --template cra/default-js --force --junit` locally succsefully (failed for registry publish but it built well)

And regarding the [test workflow,](https://circleci.com/workflow-run/3b8ede17-fb29-431b-a661-49101e396198?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) I have no premission to watch the test suit but i ran it locally with no issues as well
image

Any ideas how to manage?

File: code/lib/ui/package.json (hunk 2 above)
Comment (JReinhold) [reply code/lib/ui/package.json]: We're in the middle of a bigger migration to `tsup` with all the packages, and one of the steps in that migration is moving most if not all dependencies to devDependencies. We're not done yet, maybe that's what you're seeing.

https://github.com/storybookjs/storybook/issues/18732

**Comment by JReinhold** [comment]
Thanks for the changes. It all looks good but I want to ensure the failing tests are okay before we merge.

@tmeasday do you think it's because when we build storybooks and sandboxes in CI we pull down the published modules but reuse the local dependencies. That would result in our issue here, where transitive dependencies of the sandboxes are looking for `@storybook/semver` but it isn't there because we've removed it from the current dependencies in this unpublished branch. That's the only explanation I have, but maybe there's something else going on.

**Comment by tmeasday** [comment]
@JReinhold I'm not 100% sure I'm understanding exactly what you mean, so.. maybe yes?

But in the interests of clarity, I think the problem is these lines:

https://github.com/storybookjs/storybook/blob/e56234254672be91060e85529a0853b75c3fe6fe/scripts/verdaccio.yaml#L103-L107

What that means is that any `@storybook/X` package that a sandbox (in `--no-link` mode[1]) tries to install must be published to the local registry.

I'm a little hazy on this; I remember running into some issues with this when I setup the sandboxes, but I don't remember what I did to resolve; perhaps I just got rid of the offending non-monorepo `@storybook/X` package somehow. I would advise doing the same here also.

[1] Which is what CI does.

**Review by ndelangen — APPROVED** [review]

**Review by JReinhold — APPROVED** [review]

**Comment by JReinhold** [comment]
@danielamram I made some changes to make the tests pass, I hope that it's okay. There were one place in the iframe webpack config that we missed because it was just called `'semver'`, so you couldn't possibly have known.

Great job! 💪

**Comment by ndelangen** [comment]
@danielamram I am trying to werge in `next` branch, but GitHub won't let me:
```
(replace-storybook-semver)⚡ % git push                                                                 ~/Projects/Storybook/core/code
Enumerating objects: 96, done.
Counting objects: 100% (96/96), done.
Delta compression using up to 10 threads
Compressing objects: 100% (34/34), done.
Writing objects: 100% (34/34), 5.29 KiB | 5.29 MiB/s, done.
Total 34 (delta 31), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (31/31), completed with 30 local objects.
To https://github.com/danielamram/storybook.git
 ! [remote rejected]       replace-storybook-semver -> replace-storybook-semver (refusing to allow an OAuth App to create or update workflow `.github/workflows/generate-repros-next.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/danielamram/storybook.git'
```

Could you please merge in `next` branch from the upstream remote? Then we can get this merged! 

Thank you for contributing!

**Comment by IanVS** [comment]
>The lockfile would have been modified by this install, which is explicitly forbidden.

@danielamram would you mind running `yarn install` in `/code` and checking in the changes to `yarn.lock`?

**Pull request merged**